win32/src/winapi/kernel32/file.rs

```diff
@@ -294,6 +438,66 @@ pub fn GetConsoleMode(
     true
 }
 
+#[win32_derive::dllexport]
+pub fn GetFullPathNameA(
+    machine: &mut Machine,
+    lpFileName: Option<&str>,
+    nBufferLength: u32,
+    lpBuffer: u32,
+    lpFilePart: Option<&mut u32>,
+) -> u32 {
+    let Some(file_name) = lpFileName else {
+        log::warn!("GetFullPathNameA failed: null lpFileName");
+        SetLastError(machine, ERROR_INVALID_DATA);
+        return 0;
+    };
+
+    let out_path = match machine.host.canonicalize(file_name) {
+        Ok(value) => value,
+        Err(code) => {
```

Comment:
Is this a `let Some(out_path) = ... else { ... }` candidate too?

Reply:
I can't access the error code with let-else, so I don't think so.